TorchSharp/THMemoryFile.cs

```diff
@@ -0,0 +1,193 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+using System.Text;
+
+namespace Torch.IO
+{
+    /// <summary>
+    /// Bindings for the native THMemoryFile APIs
+    /// </summary>
+    public class MemoryFile : File {
+
+        /// <summary>
+        ///    The storage class provides a mechanism to access the underlying data representation for memory files.
+        /// </summary>
+        public class CharStorage : IDisposable {
```

Comment:
So my understanding is that CharStorage is used here while the other type of storages are used to back a tensor. Is this correct?

Reply:
Yes, that's what the C interface is using for memory files.

Reply:
Storages now are in 2 different files. Should we do as in Torch where all storages classes are in one file? This can be solved in a separate PR (related to #50?)